### PR TITLE
updated contentvalidator to explicitly look at conanfile.py

### LINKED ISSUE #35: Upload of packages with shebang line in conanfile.py fails with strict content type validation on
First off, great work on this!

* **What are you trying to do?**

An example way to reproduce the issue, using Conan client 1.4.5 (also verified with >1.7), nexus-repository-conan 0.0.3, and Nexus OSS version 3.12.1-01. 

1. `$> conan install boost_algorithm/1.66.0@bincrafters/stable --remote=bincrafters-public-conan` (where bincrafters-public-conan is https://api.bintray.com/conan/bincrafters/public-conan, and the particular choice of `boost_algorithm` is important because the `conanfile.py` for it has a [shebang line](https://www.in-ulm.de/~mascheck/various/shebang/) on top, i.e. `#!/usr/bin/env python`)
2. `$> conan remote add conan-internal <MY-URL>` 
3. `$> conan user username -p password -r conan-internal`
4. `$> conan upload boost_algorithm/1.66.0@bincrafters/stable --remote=conan-internal --all`
5. Output:
```
Uploading boost_algorithm/1.66.0@bincrafters/stable to remote 'conan-internal'
Uploading conanmanifest.txt
[==================================================] 103B/103B
Uploading conanfile.py
[==================================================] 2.6KB/2.6KB
ERROR:
Error uploading file: conanfile.py, 'b'''
Uploading conan_export.tgz
[==================================================] 758B/758B
ERROR: Execute upload again to retry upload the failed files: conanfile.py. [Remote: conan-internal]
```
After placing a breakpoint [here](https://github.com/conan-io/conan/blob/399d2366533f2aeb4b1c4a63eeb83453bfcc8e2e/conans/client/rest/rest_client.py#L486) (for Conan client version 1.4.5, it moved in later versions but same issue), `response` was `HTTP 400 Bad Request: 'Detected content type [application/x-sh], but expected [text/x-python]: /v1/conans/bincrafters/boost_algorithm/1.66.0/stable/conanfile.py'`. This [unrelated SO post](https://stackoverflow.com/questions/50182301/deploying-a-jar-artifact-to-nexus-results-in-a-content-type-mismatch) tipped me off to disabling Strict Content Type Validation on the repository to work around this, and that worked, I was able to upload after unchecking this (see image below)
strict_content_validation

Given this behavior, I feel like the issue is rooted in the shebang line, but I don't understand enough about the content type validation feature to know whether it's a good idea to turn it off just for this. 

* **What feature or behavior is this required for?**

Any Conan package that has a shebang line won't be able to be uploaded without disabled strict content validation on the repository.

* **How could we solve this issue? (Not knowing is okay!)**

Obviously telling folks to uncheck Strict Content Type Validation seems to be a viable solution, but again, not sure if that's really the right answer. 

* **Anything else?**

### PR DESCRIPTION
**Pull request by doddi**
The issue is as explained on the original ticket.
This PR explicitly checks for /conanfile.py rather than any .py file and returns the correct content type

* Fixes #35

### CODE REVIEW

File: src/main/java/org/sonatype/repository/conan/internal/security/ConanContentValidator.java [hunk 1]
```diff
@@ -42,8 +44,13 @@ public String determineContentType(final boolean strictContentTypeValidation,
                                      @Nullable final String contentName,
                                      @Nullable final String declaredContentType) throws IOException
   {
-    if (contentName != null && contentName.endsWith(".tgz")) {
-      return ContentTypes.APPLICATION_GZIP;
+    if (contentName != null) {
+      if (contentName.endsWith(".tgz")) {
+        return ContentTypes.APPLICATION_GZIP;
+      }
+      else if (contentName.endsWith("/conanfile.py")) {
```
Comment (jlstephens89) [inline src/main/java/org/sonatype/repository/conan/internal/security/ConanContentValidator.java]: Question: Why does limiting this just to the conanfile fix the problem? Shouldn't the other files also be identified as python files?

Comment (doddi) [reply src/main/java/org/sonatype/repository/conan/internal/security/ConanContentValidator.java]: As far as I am aware there is only a conanfile.py as part of the packaging which is why I wanted to limited it to just this file.

File: src/test/java/org/sonatype/repository/conan/internal/security/ConanContentValidatorTest.java [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+
+  @Before
+  public void setUp() throws Exception {
+    DefaultContentValidator defaultContentValidator = new DefaultContentValidator(new DefaultMimeSupport());
+
+    underTest = new ConanContentValidator(defaultContentValidator);
+  }
+
+  @Test
+  public void shouldBePythonFileWhenUsingSheBang() throws IOException {
```
[the remark this answers is not in the document]
Comment (doddi) [reply src/test/java/org/sonatype/repository/conan/internal/security/ConanContentValidatorTest.java]: As far as I am aware there isnt any non confile.py files

[the remark this answers is not in the document]
Comment (jlstephens89) [reply src/test/java/org/sonatype/repository/conan/internal/security/ConanContentValidatorTest.java]: I know but I'm suggesting we test the non-happy path

File: src/test/resources/org/sonatype/repository/conan/internal/security/conanfile.py [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+        license_contents = tmp[2:tmp.find("*/", 1)]
+        tools.save("LICENSE", license_contents)
+
+    def package(self):
+        self.copy("*.hpp", dst="include")
+        self.copy("LICENSE")
+
+    def package_info(self):
+        self.cpp_info.libdirs = []
+        self.cpp_info.bindirs = []
```
[the remark this answers is not in the document]
Comment (doddi) [reply src/test/resources/org/sonatype/repository/conan/internal/security/conanfile.py]: https://github.com/sonatype-nexus-community/nexus-repository-conan/pull/41/commits/f40e45c612b6550043d23e7b746c47abd7e25ede

File: src/test/java/org/sonatype/repository/conan/internal/security/ConanContentValidatorTest.java [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+    String contentType = underTest.determineContentType(true,
+        () -> getClass().getResourceAsStream("conanfile.py"),
+        MimeRulesSource.NOOP,
+        "something/conanfile.py",
+        "text/x-python");
+
+    assertThat(contentType, is("text/x-python"));
+  }
+}
```
[the remark this answers is not in the document]
Comment (doddi) [reply src/test/java/org/sonatype/repository/conan/internal/security/ConanContentValidatorTest.java]: https://github.com/sonatype-nexus-community/nexus-repository-conan/pull/41/commits/f40e45c612b6550043d23e7b746c47abd7e25ede

File: src/test/java/org/sonatype/repository/conan/internal/security/ConanContentValidatorTest.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (doddi) [reply src/test/java/org/sonatype/repository/conan/internal/security/ConanContentValidatorTest.java]: gotcha https://github.com/sonatype-nexus-community/nexus-repository-conan/pull/41/commits/a9688ce648e3a490d941cce30f9d061a1dde3450

**Pull request merged**